app/shared/src/commonMain/kotlin/ui/subject/episode/EpisodeViewModel.kt

```diff
@@ -663,14 +679,23 @@ class EpisodeViewModel(
             }
         },
         onSend = { context, content ->
-            when (context) {
-                is CommentContext.Episode ->
-                    bangumiCommentService.postEpisodeComment(episodeId.value, content)
-
-                is CommentContext.Reply ->
-                    bangumiCommentService.postEpisodeComment(episodeId.value, content, context.commentId)
-
-                is CommentContext.SubjectReview -> {} // TODO: send subject comment
+            val token = suspend {
+                withContext(Dispatchers.Main) { turnstileState.reloadAndGetToken() }
+            }
+                .asFlow()
+                .retry(3)
+                .catch { logger.error(it) { "Failed to get turnstile token" } }
+                .firstOrNull()
+
+            if (token == null) return@CommentEditorState false
+
+            try {
+                bangumiCommentRepository.postComment(context, content, token)
+                commentStateRestarter.restart() // 评论发送成功了, 刷新一下
+                return@CommentEditorState true
+            } catch (e: RepositoryException) {
+                logger.error(e) { "Failed to post comment, see exception" }
+                return@CommentEditorState false
```

Comment:
don't print all exceptions (which includes NetworkException). For (e !is RepositoryException) you should just print its `toString()`, not its stacktrace.